lib/spack/spack/schema/config.py

```diff
@@ -71,6 +71,10 @@
             "license_dir": {"type": "string"},
             "source_cache": {"type": "string"},
             "misc_cache": {"type": "string"},
+            "enable_concretization_cache": {"type": "boolean"},
+            "concretization_cache_path": {"type": "string"},
+            "concretization_cache_entry_limit": {"type": "integer", "minimum": 1},
+            "concretization_cache_byte_limit": {"type": "integer", "minimum": 800},
```

Comment:
Why not using a sub-object in the configuration? It would be:
```yaml
concretization_cache:
    enable: true
    path: ....
    entry_limit: ...   
    size_limit: ...
```
compared to:
```yaml
enable_concretization_cache: ...
concretization_cache_path: ....
concretization_cache_entry_limit: ...   
concretization_cache_byte_limit: ...
```
I would also use the generic `size` instead of `byte`, cause that wouldn't need to change if later we allow `5GB` or something similar as a valid value.

Reply:
> Is your thought the user would indicate something like 5GB as a string in their config? I'm hesitant to add extra parsing and validation for config values, but it would be nice to not have to just give the config a giant number.

Yes, what I mean is that we don't have to implement it right now, but later we can say "if it's an int, it's supposed to be bytes, otherwise we can use properly formatted strings". If we call the attribute `byte_limit` we are committed to understand the entry as bytes.

Reply:
Yeah, sorry, the `pedantic` part was me thinking that if we removed the `B` from:
```
size_limit: 5GB
```
`byte_limits` works too, cause the unit of measure is already in the attribute name :slightly_smiling_face:

Reply:
> Yeah, sorry, the pedantic part was me thinking that if we removed the B from:

No need for apology, I was/am your side (I'm also a fan of being pendantic from time to time) 😄 
That's actually really funny.


> byte_limits works too, cause the unit of measure is already in the attribute name 🙂

I honestly don't have an opinion, I'll go with whatever you/the general consensus prefers.